feat(inbox,tui): surface task-retry-pattern failures for easier mission debugging

## Observation (from M1-r2 in Phase 4 validation, 2026-04-17)

Mission \`m1-todo-v2-r2\` is 18 min in, 8 of 18 tasks done, 2 at the retry ceiling (attempts=3). Looking at \`antfarm inbox\`:

\`\`\`
[ERROR] stale_worker: ...  (good)
[WARNING] blocked_by_deps: ...  (good)
\`\`\`

But inbox does NOT surface the actual problem. To find out, an operator has to:
1. Guess which task is retrying (no aggregate view).
2. \`antfarm scent <task-id>\` or query \`/tasks/<id>\` directly.
3. Read the trail manually to find the failure reason.

For \`task-m1-todo-v2-r2-01\`, the trail showed 3 consecutive attempts each ending with:
\`tests failed: ImportError while loading conftest '...tests/conftest...'\`

A systemic issue — but invisible unless you dig.

## Gaps

### 1. Inbox doesn't aggregate retry patterns

No \`[ERROR] task_retrying: task-01 has failed 3/3 attempts. Last failure: tests failed: ImportError while loading conftest...\` entry.

Proposed: doctor check \`check_retry_patterns\` that flags any task with attempts ≥ max_attempts - 1 and summarizes the last failure reason (extracted from trail).

### 2. No "why is this mission slow" diagnostic

No single command like \`antfarm mission diagnose <id>\` that returns:
- Task count by status.
- Retry distribution (e.g., \"5 tasks retried ≥2 times\").
- Top 3 common failure reasons across all tasks.
- Tasks blocked by dependencies, with blocker chain.

### 3. TUI Activity column is empty under claude-code

All 6 workers in the current M1 run show \`status=idle action=\` in the TUI's \`/status/full\` response — even while they're clearly running Claude Code subprocesses. The v0.7.0 Activity column (#245) was supposed to show \`tool:Bash\` / \`tool:Edit\` etc., but the Claude Code adapter apparently never fires the activity hooks that \`/workers/{id}/activity\` consumes.

Proposed: verify the Claude Code adapter's PostToolUse hook (\`antfarm/adapters/claude_code/hooks/heartbeat.sh\` and any activity hook) posts to \`/workers/{id}/activity\` on every tool call.

## Why this matters

When a mission is in trouble, the operator needs to answer \"what's wrong and what do I do\" in under 30 seconds. Right now it takes several manual queries. That friction will bite hard on Brahmaos or any real project.

## Severity

Medium. Not a correctness bug — data is all there, just hard to surface. But for Phase 4 validation and Brahmaos readiness, self-serve operator UX matters a lot.

## Source

Discovered during Phase 4 M1-r2. Related to #320 (autoscaler depth-aware) and #322 (queen parallelism) — all Phase-5-class improvements that surfaced during mission validation.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "antfarm"
-version = "0.6.7"
+version = "0.6.8"
 description = "Lightweight orchestration layer for distributing coding work across multiple AI coding agents"
 readme = "README.md"
 license = {text = "MIT"}

